server/listeners/transform-event.js

```diff
@@ -28,6 +28,32 @@ const getReferences = (payload, messageTypes) => {
   return [];
 };
 
+const getBy = (payload) => { //eslint-disable-line
+  let by = [];
+  if (_.isArray(payload.by)) {
+    for (let item of payload.by) {
+      if (item.endpoint && item.endpoint.id) {
+        by.push({ endpoint: { id: item.endpoint.id, token: item.endpoint.token } });
+      }
+      if (item.user && item.user.id) {
+        by.push({ user: { id: item.user.id, token: item.user.token } });
+      }
+    }
+  } else if (payload.by) {
+    if (payload.by.user && payload.by.user.id) {
+      by.push({ user: { id: payload.by.user.id, token: payload.by.user.token } });
+    }
+    if (_.isArray(payload.by.endpoints)) {
+      for (let item of payload.by.endpoints) {
```

Comment:
I also didn't see how this would contain more than one item? It always looks to just be one item as an object rather than an array.